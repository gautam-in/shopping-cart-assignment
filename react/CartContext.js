import React from 'react';

// set the defaults
const CartContext = React.createContext({
  cart: [],
  setCart: () => {},
});

export default CartContext;
