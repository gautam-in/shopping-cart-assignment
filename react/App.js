import React, { useState } from 'react';
import Home from './Components/home';
import Products from './Components/products';
import Login from './Components/login';
import Register from './Components/register';
import Nav from './Components/Nav';
import './Styles/main.css';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import CartContext from './CartContext';

export default function () {
  const [cart, setCart] = useState([]);
  const value = { cart, setCart };

  return (
    <CartContext.Provider value={value}>
      <Router>
        <header className="header">
          <Nav />
        </header>
        <main>
          <Switch>
            <Route path="/login">
              <Login />
            </Route>
            <Route path="/register">
              <Register />
            </Route>
            <Route path="/products/:categoryId">
              <Products />
            </Route>
            <Route path="/products">
              <Products />
            </Route>
            <Route path="/">
              <Home />
            </Route>
          </Switch>
        </main>
      </Router>
    </CartContext.Provider>
  );
}
