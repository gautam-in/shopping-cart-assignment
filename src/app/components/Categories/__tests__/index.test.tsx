import * as React from 'react';
import { render } from '@testing-library/react';

import { Categories } from '..';

jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: str => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
      react: { useSuspense: false },
    };
  },
}));

describe('<Categories  />', () => {
  it('should match snapshot', () => {
    const loadingIndicator = render(<Categories loading={false} items={[]} />);
    expect(loadingIndicator.container.firstChild).toMatchSnapshot();
  });
});
