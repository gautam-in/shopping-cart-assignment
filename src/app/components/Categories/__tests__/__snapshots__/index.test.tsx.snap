// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Categories  /> should match snapshot 1`] = `<div />`;
